util/keyfile/src/keygen.rs

```diff
@@ -4,30 +4,54 @@
 //! corresponding to `mc_account_keys::RootIdentity`, and
 //! `mc_account_keys::PublicAddress` respectively.
 
-use crate::{read_keyfile, read_pubfile, write_keyfile, write_pubfile};
+use crate::{
+    error::Error, read_keyfile, read_pubfile, read_root_entropy_keyfile, write_keyfile,
+    write_pubfile,
+};
+use bip39::{Language, Mnemonic};
 use mc_account_keys::{AccountKey, PublicAddress, RootIdentity};
-use rand::SeedableRng;
-use rand_hc::Hc128Rng as FixedRng;
+use mc_account_keys_slip10::Slip10KeyGenerator;
+use rand::{RngCore, SeedableRng};
+use rand_hc::Hc128Rng;
 use std::{
     cmp::Ordering,
     ffi::OsStr,
     fs,
     path::{Path, PathBuf},
 };
 
-pub const DEFAULT_SEED: [u8; 32] = [1; 32];
-
-// Write a single pair of keyfiles using a given name and data
+/// Write a single pair of keyfiles using a given name and data
 pub fn write_keyfiles<P: AsRef<Path>>(
     path: P,
     name: &str,
-    root_id: &RootIdentity,
-) -> Result<(), std::io::Error> {
-    let acct_key = AccountKey::from(root_id);
+    mnemonic: &Mnemonic,
+    account_index: u32,
+    fog_report_url: Option<&str>,
+    fog_report_id: Option<&str>,
+    fog_authority_spki: Option<&[u8]>,
+) -> Result<(), Error> {
+    let slip10key = mnemonic.clone().derive_slip10_key(account_index);
+    let acct_key =
+        if fog_report_url.is_some() && fog_report_id.is_some() && fog_authority_spki.is_some() {
```

Comment:
This might be too nitty but I would consider changing this into a match statement that has `(Some, Some Some)`, `(None, None, None)`, and `_` cases, having the catch-all one panic. This could prevent someone from supplying only some of the fog arguments and not realizing they are being ignored.